core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -1175,14 +1176,23 @@ public static GroupingSetDescriptor singleGroupingSet(List<String> groupingKeys)
 
     public static class DynamicFilterPattern
     {
-        private final SymbolAlias probe;
+        private final Expression probe;
         private final ComparisonExpression.Operator operator;
         private final SymbolAlias build;
         private final boolean nullAllowed;
 
-        public DynamicFilterPattern(String probeAlias, ComparisonExpression.Operator operator, String buildAlias, boolean nullAllowed)
+        public DynamicFilterPattern(String probeExpression, ComparisonExpression.Operator operator, String buildAlias, boolean nullAllowed)
         {
-            this.probe = new SymbolAlias(requireNonNull(probeAlias, "probeAlias is null"));
+            this(
+                    rewriteIdentifiersToSymbolReferences(new SqlParser().createExpression(probeExpression, new ParsingOptions())),
```

Comment:
nit: it seems that this pattern is used in couple of places, we could have a dedicated utility method for it

Reply:
actually, there is already a method PlanBuilder.expression (maybe we should move to ExpressionTestUtils?).
I used it here + refactored other places.